chain/rpc/src/rpc.rs

```diff
@@ -110,26 +114,43 @@ impl<P: JsonRpcClient + 'static> RpcOperations<P> {
             provider,
         })
     }
+
+    pub(crate) async fn get_block_number(&self) -> Result<u64> {
+        Ok(self
+            .provider
+            .get_block_number()
+            .await?
+            .as_u64()
+            .saturating_sub(self.cfg.finality as u64))
```

Comment:
But this can never be correct for blocks with indices 0..finality.

Reply:
What do you suggest then? `finality.max(block_number)` ?
All options are wrong here.

Reply:
Should it then return `Option<u64>`?

Reply:
But it does not matter for real chains, because those are already started. So disregard.